src/DynamoCore/Core/DynamoMigrator.cs

```diff
@@ -88,11 +122,98 @@ protected virtual FileVersion DynamoVersion
             }
         }
 
+        #endregion
+
         protected DynamoMigratorBase(string rootFolder)
         {
             this.rootFolder = rootFolder;
         }
 
+        #region virtual methods
+
+        /// <summary>
+        /// Reads preference settings from input file and deserializes it into 
+        /// PreferenceSettings object in memory
+        /// This function can be overridden by version specific migrator classes
+        /// </summary>
+        /// <param name="sourceMigrator"> source migrator version from which to migrate from </param>
+        /// <returns> returns the deserialized settings object </returns>
+        protected virtual PreferenceSettings WritePreferences(DynamoMigratorBase sourceMigrator)
+        {
+            string preferenceSettingsFilePath = sourceMigrator.PreferenceSettingsFilePath;
+            PreferenceSettings settings = null;
+            if (string.IsNullOrEmpty(preferenceSettingsFilePath))
+                return null;
+
+            try
+            {
+                var serializer = new XmlSerializer(typeof(PreferenceSettings));
+
+                using (var fs = new FileStream(preferenceSettingsFilePath, FileMode.Open, FileAccess.Read))
+                {
+                    settings = serializer.Deserialize(fs) as PreferenceSettings;
+                    fs.Close(); // Release file lock
+                }
+            }
+            catch (Exception ex)
+            {
+                throw new Exception(ex.Message);
+            }
```

Comment:
Please remove this `try-catch` block, it's catching and throwing the same exception.